scripts/README_DEV.md

```diff
@@ -1,42 +1,117 @@
-2. Get your Docker IP address with `boot2docker ip`
-3. Add a line to your /etc/hosts that goes `<Docker IP> dockerhost` 
-4. Run `boot2docker shellinit` and copy the export statements to your ~/.bash_profile 
+1. Download and set up Boot2Docker
+	1. Follow the instructions at http://docs.docker.com/installation/mac/
+		1. Use the Boot2Docker command-line utility
+		2. If you do command-line setup use: `boot2docker init eval “$(boot2docker shellinit)”`
+	2. Get your Docker IP address with `boot2docker ip`
+	3. Add a line to your /etc/hosts that goes `<Docker IP> dockerhost`
+	4. Run `boot2docker shellinit` and copy the export statements to your ~/.bash_profile
+2. Download Go from http://golang.org/dl/
+3. Set up your Go workspace
+	1. `mkdir ~/go`
+	2. Add the following to your ~/.bash_profile
+
+		``` bash
+		export GOPATH=$HOME/go
+		export PATH=$PATH:$GOPATH/bin
+		```
+	3. Reload your bash profile
+		1. `source ~/.bash_profile`
+4. Install Node.js using Homebrew
+	1. Download Homebrew from http://brew.sh/
+	2. `brew install node`
+5. Install Compass
+	1. Make sure you have the latest verison of Ruby
+	2. `gem install compass`
+6. Install Mercurial (needed to use Godep)
+	1. `brew install mercurial`
+7. Download Mattermost
+
+	``` bash
+	cd ~/go
+	mkdir -p src/github.com/mattermost
+	cd src/github.com/mattermost
+	git clone github.com/mattermost/platform.git
+	cd platform
+	```
+8. Run unit tests on Mattermost using `make test` to make sure the installation was successful
+9. If tests passed, you can now run Mattermost using `make run`
 
 Any issues? Please let us know on our forums at: http://forum.mattermost.org
 
-GO SETUP
+### Ubuntu ###
+
+1. Download Docker
+	1. Follow the instructions at https://docs.docker.com/installation/ubuntulinux/ or use the summary below:
 
-1. Download Go from http://golang.org/dl/ 
+		``` bash
+		sudo apt-get update
+		sudo apt-get install wget
+		wget -qO- https://get.docker.com/ | sh
+		sudo usermod -aG docker <username>
+		sudo service docker start
+		newgrp docker
+		```
+2. Install build essentials
+	1. `apt-get install build-essential`
+3. Download Go from http://golang.org/dl/
+4. Set up your Go workspace and add Go to the PATH
+	1. `mkdir ~/go`
+	2. Add the following to your ~/.bashrc
 
-NODE.JS SETUP 
+		``` bash
+		export GOPATH=$HOME/go
+		export GOROOT=/usr/local/go
+		export PATH=$PATH:$GOROOT/bin
+		```
+	3. Reload your bashrc
+		1. `source ~/.bashrc`
+5. Install Node.js
+	1. Download the newest version of the Node.js sources from https://nodejs.org/download/
+	2. Extract the contents of the package and cd into the extracted files
+	3. Compile and install Node.js
 
-1. Install homebrew from http://brew.sh 
-2. `brew install node`
+		``` bash
+		./configure
+		make
+		make install
+		```
+6. Install Ruby and Compass
 
-COMPASS SETUP 
+	``` bash
+	apt-get install ruby
+	apt-get install ruby-dev
+	gem install compass
+	```
+7. Install Mercurial (needed for Godep)
```

Comment:
That's a good point. We are still using godep to run go, but that's just to make it look for dependencies in _workspace. I'll see how it runs without Mercurial installed.